client/db/src/bench.rs

```diff
@@ -200,40 +212,71 @@ impl<B: BlockT> BenchmarkingState<B> {
 					};
 					key_tracker.insert(key.to_vec(), has_been_read);
 					read_write_tracker.add_read();
+					true
 				} else {
 					read_write_tracker.add_repeat_read();
+					false
 				}
 			}
+		};
+
+		if read {
+			if let Some(childtrie) = childtrie {
+				log::trace!(
+					target: "benchmark",
+					"Childtrie Read: {} {}", HexDisplay::from(&childtrie), HexDisplay::from(&key)
+				);
+			} else {
+				log::trace!(target: "benchmark", "Read: {}", HexDisplay::from(&key));
+			}
 		}
 	}
 
-	fn add_write_key(&self, key: &[u8]) {
-		log::trace!(target: "benchmark", "Write: {}", HexDisplay::from(&key));
-
-		let mut key_tracker = self.key_tracker.borrow_mut();
+	// Childtrie is identified by its storage key (i.e. `ChildInfo::storage_key`)
+	fn add_write_key(&self, childtrie: Option<&[u8]>, key: &[u8]) {
 		let mut read_write_tracker = self.read_write_tracker.borrow_mut();
+		let mut child_key_tracker = self.child_key_tracker.borrow_mut();
+		let mut main_key_tracker = self.main_key_tracker.borrow_mut();
```

Comment:
Would it be better to only borrow this value in appropriate branch the if statement below?

Reply:
actually if we borrow inside the branch then the borrow doesn't live long enough